developer/src/kmc-kmn/src/kmw-compiler/javascript-strings.ts

```diff
@@ -620,8 +620,8 @@ begin*/
 }
 
 function isGroupReadOnly(fk: KMX.KEYBOARD, fgp: KMX.GROUP) {
-  // TODO: export group + store metadata in debug store, use that
-  return false;
+  const index = fk.groups.indexOf(fgp);
+  return kmxResult.extra.groups[index].isReadOnly;
```

Comment:
Does this return false if `fk.groups` is `[]`?

Reply:
I think that any keyboard should already have at least one group - the main one.  There's no keystroke processing without one.

Still... I suppose it doesn't hurt to add a null-guard?

Reply:
In the compiler, `fgp` is guaranteed to be a member of the `fk.groups` array -- we never construct isolated `KMX.GROUP` objects; they are read directly from the .kmx. Also, the `extra.groups` array is guaranteed to have a 1:1 correspondence to the `fk.groups` array, as both are returned by the .kmx compiler.